Automattic/wp-calypso
# A4A > Feedback: Logical implementation of in-product feedback mechanism

Related to https://github.com/Automattic/automattic-for-agencies-dev/issues/1246

## Proposed Changes

This PR makes the logical implementation of the in-product feedback mechanism.

## Why are these changes being made?


* To show the feedback form after specific actions are performed: 

• Sending a referral
• Adding a team member
• Filling out partner directory information

NOTE: The UI is not yet integrated with the API endpoint, as it is in progress.

## Testing Instructions

* Open the A4A live link
* Append the URL with ?flags=-a4a-product-feedback, perform the below actions and verify no feedback form is shown. Please ensure you don't refresh the page, as it will add the feature flag back. 

• Sending a referral (/marketplace/products > Toggle the refer mode on and send a referral)
• Adding a team member (/team/invite > Send invite)
• Filling out partner directory information (/partner-directory/agency-details > Save public profile)

* Remove the feature flag > Refresh the page and perform the below actions and verify the feedback is shown as displayed below. Submitting or skipping the form should hide it and not show again when performing the same action. You can verify it by visiting the page manually with `#feedback` appended, you can find the URLs below. 

• Sending a referral (/marketplace/products > Toggle the refer mode on and send a referral)

Manual testing URL: /referrals/dashboard#feedback

<img width="1728" alt="Screenshot 2024-12-05 at 10 28 38 AM" src="https://github.com/user-attachments/assets/8f11e222-8be1-4257-921a-4ab69124a8e8">

• Adding a team member (/team/invite > Send invite)

Manual testing URL: /team/invite#feedback

<img width="1728" alt="Screenshot 2024-12-05 at 10 12 24 AM" src="https://github.com/user-attachments/assets/07755c1e-c1a6-432b-a332-016b792d5311">

• Filling out partner directory information (/partner-directory/agency-details > Save public profile)

Manual testing URL: /partner-directory/agency-details#feedback

<img width="1728" alt="Screenshot 2024-12-05 at 10 28 01 AM" src="https://github.com/user-attachments/assets/28c80620-6643-4b8c-8544-f718e6b82e56">

How the preferences are stored:

<img width="304" alt="Screenshot 2024-12-05 at 11 06 08 AM" src="https://github.com/user-attachments/assets/1ca12427-cee1-4f6c-9a3c-c40272e94c6a">

Preferences can be reset from staging


## Pre-merge Checklist
